tests/scripting_test.cpp

```diff
@@ -45,23 +47,23 @@ BOOST_AUTO_TEST_CASE(TestGetProvider)
 
     PluginLoader::Instance()->loadService("scripting",tc);
 
-    // We use a sequential activity in order to force execution on trigger().
-    tc->stop();
-    BOOST_CHECK( tc->setActivity( new SequentialActivity() ) );
-    tc->start();
-
     boost::shared_ptr<Scripting> sc = tc->getProvider<Scripting>("scripting");
     BOOST_REQUIRE( sc );
     BOOST_CHECK ( sc->ready() );
-    bool r;
-    r = sc->loadProgramText("program Foo { do test.assert(true); set ret = 10.0; }");
-    BOOST_REQUIRE( r );
+    BOOST_REQUIRE( sc->loadProgramText( MULTILINE_STRING(
+        program Foo {
+            do test.assert(true);
+            set ret = 10.0;
+        })));
```

Comment:
This is equally readable, and does not introduce foreign code into program (which could screw up syntax highlighters, ctags or similar symbol tracking devices,  etc.)

```C++
BOOST_REQUIRE( sc->loadProgramText(
    "program Foo {"
    "    do test.assert(true);"
    "    set ret = 10.0;"
    "}"));
```

Reply:
I personally would prefer the preprocessor stringification and there is no foreign code except for the additional preprocessor macro defined in the file header. Especially if the script snippet contains string literals the need to escape quotes reduces readability.

QtCreator perfectly understands C preprocessor stringification and does not complain about missing brackets or similar syntax errors, but still highlights keywords also known in C/C++.